extras/build/material-icons.js

```diff
@@ -1,48 +1,109 @@
+  baseline: '', // filled
+  outlined: '_outlined',
+  round: '_round',
+  sharp: '_sharp',
+}
 
-svgFiles.forEach(file => {
-  const name = ('mat_' + file.match(/ic_(.*)_24px\.svg/)[1])
-    .replace(/(_\w)/g, m => m[1].toUpperCase())
+function downloadIcon(icon) {
+  return Promise.all(
+    Object.keys(themeMap).map(async (theme) => {
+      const themeName = themeMap[theme]
+      const formattedTheme = themeName.split('_').join('')
+      const response = await fetch(
+        `https://fonts.gstatic.com/s/i/materialicons${formattedTheme}/${icon.name}/v${icon.version}/24px.svg`,
+      )
+      if (response.status !== 200) {
+        throw new Error(`status ${response.status}`)
+      }
+      const SVG = await response.text()
+      const name = ((themeName === '' ? 'mat_' : theme + '_') + icon.name)
+        .replace(/(_\w)/g, m => m[1].toUpperCase())
 
-  if (iconNames.has(name)) {
-    return
-  }
+      if (iconNames[theme].has(name)) {
+        return
+      }
+
+      try {
+        const { svgDef, typeDef } = extractSvg(SVG, name)
+        svgExports[theme].push(svgDef)
+        typeExports[theme].push(typeDef)
 
+        iconNames[theme].add(name)
+      }
+      catch(err) {
+        console.error(err)
+        skipped[theme].push(name)
+      }
+    }),
+  )
+}
+
+async function run() {
   try {
-    const { svgDef, typeDef } = extract(file, name)
-    svgExports.push(svgDef)
-    typeExports.push(typeDef)
+    const response = await fetch('https://fonts.google.com/metadata/icons')
+    const text = await response.text()
+    const data = JSON.parse(text.replace(")]}'", ''))
+    let icons = data.icons
+    icons = icons.map((icon, index) => ({ index, ...icon }))
 
-    iconNames.add(name)
-  }
-  catch(err) {
-    console.error(err)
-    skipped.push(name)
+    console.log('\nDownloading Google Material Design SVGs...')
+    console.log(`${icons.length} icons to download...`)
+
+    Object.keys(themeMap).map(async (theme) => {
+      if (skipped[theme] === void 0) skipped[theme] = []
+      if (svgExports[theme] === void 0) svgExports[theme] = []
+      if (typeExports[theme] === void 0) typeExports[theme] = []
+      if (iconNames[theme] === void 0) iconNames[theme] = new Set()
+      if (distFolder[theme] === void 0) distFolder[theme] = resolve(__dirname, `../material-icons${themeMap[theme]}`).replace(/_/g, '-')
+    })
+
+    const queue = new Queue(
+      async (icon) => {
+        await retry(async ({ tries }) => {
+          await sleep((tries - 1) * 100)
+          await downloadIcon(icon)
+        })
+      },
+      { concurrency: 5 },
+    )
+    queue.push(icons)
+    await queue.wait({ empty: true })
+
+    console.log('')
+    Object.keys(themeMap).map(async (theme) => {
+      console.log((`Updating SVG for ../material-icons${themeMap[theme]}`).replace(/_/g, '-'))
+      writeExports(iconSetName, packageName, distFolder[theme], svgExports[theme], typeExports[theme], skipped[theme])
+
+      // copySync(
```

Comment:
So we'll go by manually copying this License, right?

Reply:
If it changes, yes. Otherwise it's already in Git and will remain static.